app/controllers/posts_controller.rb

```diff
@@ -1,23 +1,64 @@
 class PostsController < ApplicationController
-  before_action :find_user, only: %i[index show]
-  before_action :find_post, only: [:show]
+  before_action :find_user, only: %i[show new create like]
+  before_action :find_post, only: %i[show like]
 
+  # GET /posts or /users/:user_id/posts
   def index
-    @posts = @user.posts
+    if params[:user_id]
+      @user = User.find_by(id: params[:user_id])
+      if @user
+        @posts = @user.posts
+      else
+        flash[:alert] = 'User not found.'
+        redirect_to root_path and return
+      end
+    else
+      @posts = Post.all
+    end
   end
 
+  # GET /users/:user_id/posts/:id
   def show; end
 
+  # GET /users/:user_id/posts/new
+  def new
+    @post = @user.posts.build
+  end
+
+  # POST /users/:user_id/posts
+  def create
+    @post = @user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  # POST /users/:user_id/posts/:id/like
+  def like
+    @post.increment!(:likes_counter)
+    redirect_to [@user, @post], notice: 'You liked a post!'
+  end
+
```

Comment:
- [ ] The like feature should not be implemented here. Like is a different resource and should have its own controller, with the `create` action. Kindly remove this method from here.